apps/qubit/modules/settings/actions/ldapAction.class.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+class SettingsLdapAction extends DefaultEditAction
+{
+  // Arrays not allowed in class constants
+  public static
+    $NAMES = array(
+      'ldapHost',
+      'ldapPort',
+      'ldapBaseDn');
+
+  protected function earlyExecute()
+  {
+
+  }
+
+  protected function addField($name)
+  {
+    switch ($name)
+    {
+      case 'ldapHost':
+      case 'ldapPort':
+      case 'ldapBaseDn':
+        $default = ($name == 'ldapPort') ? 389 : '';
+        if (null !== $this->{$name} = QubitSetting::getByName($name))
```

Comment:
Is this an optimization to save extra queries later when POSTing? If that's the case, you may actually don't need it because `BaseSetting` is caching populated objects for you here: https://github.com/artefactual/atom/blob/qa/2.3.x/lib/model/om/BaseSetting.php#L32. You may want to double check though.